### ISSUE TITLE
Support React 18

### ISSUE THREAD

**Issue by juandgarin** [Azure/communication-ui-library#1900]
When i try to install the library, it returns me an error indicating that there are dependencies that cant resolve.

Anyone with the same problem?

I think it is a React 18 issue

![Screenshot 2022-05-17 at 06 09 36](https://user-images.githubusercontent.com/62432378/168774913-9f4cbe9d-1507-4bb8-9d50-2826dcb40938.png)

**Comment by mgamis-msft** [comment]
Hi @juandgarin, thank you for filing this issue. I will look into this today.

**Comment by mgamis-msft** [comment]
Hi @juandgarin, our dependency on [fluent ui library](https://www.npmjs.com/package/@fluentui/react) restricts our package from using React 18 or higher. What is your use case for `backoffice_next@0.1.0`? Is it possible to use another version of backoffice that does not rely on React 18?

**Comment by prprabhu-ms** [comment]
Hi @juandgarin I don't expect us to be able to lift the < React18 requirement easily at the moment (Fluentv8 recently moved to React 17, and I'm not aware of any plans to move beyond React18+)

Closing this issue as it is not currently actionable, but please feel free re-open with any more information / insight you have for the need for React 18+

**Comment by prprabhu-ms** [comment]
Actually, let's keep this open to track when FluentUI goes to React 18 and then this issue will be actionable: https://github.com/microsoft/fluentui/issues/22592

**Comment by prprabhu-ms** [comment]
It is still unclear if Fluentv8 will gain React18 support at all: https://github.com/microsoft/fluentui/issues/22592#issuecomment-1113633465

**Comment by sai0909** [comment]
I am also facing the same issue when I followed these steps. https://azure.github.io/communication-ui-library/?path=/docs/quickstarts-uicomponents--page

**Comment by prprabhu-ms** [comment]
@sai0909 Can you try downgrading to React v17 to see if it fixes your issue?

**Comment by prprabhu-ms** [comment]
It is a big problem that anybody using our quickstart will currently hit the problem posted in OP.

I confirmed that downgrading the app to React 17 with

```bash
npm install react@17 react-dom@17 @testing-library/react@12 @types/react@17 @types/react-dom@17
```

almost fixes the issue.
The problem is that `react-dom` had a breaking change from v17 -> v18 so that the `create-react-app` generated boilerplate doesn't work with React 17.

If you modify the generated `index.tsx` to be 

```ts
import React from 'react';
import { render }  from 'react-dom';
import './index.css';
import App from './App';
import reportWebVitals from './reportWebVitals';

const container = document.getElementById('root');
render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  container
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
```

Then the quickstart works correctly.

But... that is *terrible* devX. Still looking for less manual workarounds.

**Comment by prprabhu-ms** [comment]
Honestly, this is a fairly large feature gap in `create-react-app`. There is _no_ way to use an older version of `react` libraries with that tool. Maybe they'll have a fix for it soon... https://github.com/facebook/create-react-app/issues/12269

**Comment by piotrzaborow** [comment]
It's not only about CRA. Next.js has their own CLI for creating new apps and usage of latest version will always bring React 18+.

I understand the complexity of migration, but not supporting new versions of React just places Fluent UI React as no-go for new React-based projects.

Of course I can downgrade, but I will lack of new Next.js features.

**Comment by prprabhu-ms** [comment]
@piotrzaborow we're following up on both forks:

* Track Fluent UI React upgrade to React 18
* Deliver a workaround that unbreaks new apps using `@azure/communication-react` right now.

I learnt that it _is_ possible to set React versions via CRA templates, so we will explore that option in the short term.

**Comment by micahgodbolt** [comment]
I'm working on React 18 support in Fluent UI right now. The changes in Strict mode are causing a good bit of churn, so we have some work ahead of us to get all of our controls working properly in React 18. For clarity, what version of Fluent UI is your package dependent on?

**Comment by micahgodbolt** [comment]
It took longer than expected, but peer deps have been updated. Both v8 and v9 will be usable within react 18 upon the next release.

**Comment by krista-m-bledsoe** [comment]
@micahgodbolt - Do you have plans or know when the communication-ui-library will be upgraded to using v9 of Fluent UI ?

**Comment by develohpanda** [comment]
Hi @micahgodbolt, while the `@fluentui/react` warning has now disappeared for me, I've still got a few others.

![image](https://user-images.githubusercontent.com/4312346/202288150-7fe803bb-c93c-4b08-a8fb-320b3c31b97a.png)

Do you know when the updated peer dependencies for `@azure/communication-react` and bundled dependency upgrades, will be published? For example, I can see that `@fluentui/react-northstar` is a fixed dependency in `@azure/communication-react` at v0.61.0, however they have published v0.65.0.

**Comment by JamesBurnside** [comment]
@develohpanda unfortunately we are blocked from permitting react 18 as a peer dependency on https://github.com/microsoft/fluentui/issues/25651

**Comment by develohpanda** [comment]
@JamesBurnside that's awesome! Please let me know if there's anything I can do to help it along. :)

**Comment by hyounoo** [comment]
Yeah the only reason our projects are held up for react 18 is this package please re consider supporting react 18

**Comment by micahgodbolt** [comment]
> Hi @micahgodbolt, while the `@fluentui/react` warning has now disappeared for me, I've still got a few others.
> 
> ![image](https://user-images.githubusercontent.com/4312346/202288150-7fe803bb-c93c-4b08-a8fb-320b3c31b97a.png)
> 
> Do you know when the updated peer dependencies for `@azure/communication-react` and bundled dependency upgrades, will be published? For example, I can see that `@fluentui/react-northstar` is a fixed dependency in `@azure/communication-react` at v0.61.0, however they have published v0.65.0.

The northstar project was originally under under Teams engineering. @jurokapsiar would be the best one to talk to about React 18 support. Do you know how critical the northstar dependency is for your package?

**Comment by develohpanda** [comment]
Presumably quite important as we are using the acs libraries to embed Teams within our site. I could try and track that dependency through the GitHub packages.

For what it’s worth, from my testing I don’t see anything immediately broken, however we do get the warnings as above. Perhaps the Northstar dependency doesn’t have anything that breaks between React 17 and React 18, but just the dependencies are listed incorrectly?

**Comment by hyounoo** [comment]
It's been almost a year since the first report.
May I ask if this has any progress?

**Comment by micahgodbolt** [comment]
I would suggest looking to see if you can remove the northstar dependency from this library. You should be able to migrate most Northstar code to v9 now.

https://react.fluentui.dev/?path=/docs/concepts-migration-from-v0-components-avatar-migration--page

**Comment by FullStackSteve** [comment]
We need React 18 support as it preventing us from moving forward to upgrade our other application to React, as we have strict policies to keep package versions consistent across our applications.

It's also preventing us from using Next.js

**Comment by jakub-guider** [comment]
This is a huge blocker. We're now stuck between requiring React 18 for our development needs, other packages, using Next.js on one side and just azure communication not supporting React 18 on the other.

**Comment by hyounoo** [comment]
> @prprabhu-ms is this still blocked?
> 
> Looks like https://github.com/microsoft/fluentui fully supports React 18 (albeit `@fluentui/react-northstar` is now discontinued so as @micahgodbolt suggests it would need to be removed from `Azure/communication-ui-library` so that React 18 can be supported)

Does it mean that they only need to update the fluent ui dependency?
Or is there some migration required?
It feels like taking forever.

**Comment by yxor** [comment]
any updates on this?

**Comment by realjesset** [comment]
i am working on lots of projects related to this library. May we know the status on this or if there is anything we can contribute to make this transition faster?

**Comment by FullStackSteve** [comment]
React 16.X EOL for Active Support was some time a go now - let us know if there is anything we can do to speed up the process of getting `communication-ui-library` to work with React `18.x` as we're starting to consider if another framework would be best as it is preventing us from using other tools (i.e. latest Next.js) in our platform

<img width="749" alt="image" src="https://user-images.githubusercontent.com/19396027/225316041-47c4010a-6ef7-4e7d-8ad4-1b1c8b309d8b.png">

**Comment by hyounoo** [comment]
> 

Please let me know if theres good framework replacing this

**Comment by hyounoo** [comment]
Does anyone working on this?

**Comment by simon-guider** [comment]
+1 this is a huge blocker for us

**Comment by hades200082** [comment]
This should have been updated before React 17 went EOL. Can't believe 18 is still not supported over a year later!

**Comment by martinesmann** [comment]
+1 this is a huge blocker. The silence and missing official statement is also a raising concern. 
Is this project about to be abandoned or has that already happened?

**Comment by hyounoo** [comment]
All the samples or sdk I use is for react 18+ but this...

**Comment by simon-guider** [comment]
@prprabhu-ms please remove the blocked label from this issue as fluent-ui has been updated to React 18 https://github.com/microsoft/fluentui/issues/22592

**Comment by JamesBurnside** [comment]
Greetings Friends, we want to thank the community for working with us and providing valuable feedback for us to understand where we should be prioritizing our work. To recap some of the previous discussion above, `@azure/communication-react` currently leverages components from `@fluentui/react-northstar`, which does not support React 18.

To address to problem, we have been working with our partner teams in Fluent over the last several months to build a migration strategy to Fluent v9. This migration will allow `@azure/communication-react` to support React 18. The main blocker we are focusing on is a direct replacement for the Chat components from `@fluentui/react-northstar`. Once that component is ready, it will unblock the migration. We will be tracking the migration and providing updates in a separate issue: https://github.com/Azure/communication-ui-library/issues/2924

However, we expect the full migration to Fluent React v9 to take some time. Therefore, to help unblock you, we are putting in motion a short-term plan mitigate the issue with React 18. Come join the discussion here: [Unblocking react 18 · Azure/communication-ui-library · Discussion #2939 · GitHub](https://github.com/Azure/communication-ui-library/discussions/2939)

We really appreciate the overall support as we work towards solving this issue!

**Comment by hyounoo** [comment]
Good news something is going on!
https://github.com/Azure/communication-ui-library/blob/jaburnsi/upgrade-to-react-18/packages/communication-react/package.json

**Comment by JamesBurnside** [comment]
Hi folks, we have released [`@azure/communication-react@1.5.1-beta.5`](https://www.npmjs.com/package/@azure/communication-react/v/1.5.1-beta.5) with react 18 support.
Please try it out and let us know if you find any issues. Pending no major issues are found this will go into our upcoming stable release.

**Comment by DercilioFontes** [comment]
Hi @JamesBurnside,

I'm testing the new version. Everything looks good but crashes in React strict mode when the ChatComposite is removed.
I do something like that:
```typescript
...
{chatAdapter && <ChatComposite
    fluentTheme={customTheme}
    onFetchAvatarPersonaData={() => Promise.resolve(avatarPersonaData)}
    options={{ topic: false }}
    adapter={chatAdapter}
  />}
...
```

And when the chat ends, it calls the chatAdapter `dispose` and sets the `chatAdapter` to `undefined`. It results in a blank screen with this error. Removing the React strict mode makes it work fine.
![853754713-image](https://github.com/Azure/communication-ui-library/assets/20172847/321c8357-a5f0-42b1-b278-86e31d588e58)

I also see this warning when it is mounted.
![Screenshot 2023-06-01 at 9 04 17 AM](https://github.com/Azure/communication-ui-library/assets/20172847/5074e963-2ab4-4341-83ac-ef3701b56300)


My stack:
    "@azure/communication-chat": "^1.3.2-beta.1",
    "@azure/communication-common": "^2.2.0",
    "@azure/communication-react": "^1.5.1-beta.5",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",

Complete errors:
Unmounting:
```
Uncaught DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
    at https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:220393:27
    at safelyCallDestroy (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:16737:13)
    at commitDeletionEffectsOnFiber (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:17558:29)
    at recursivelyTraverseDeletionEffects (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:17482:13)
    at commitDeletionEffectsOnFiber (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:17499:17)
    at recursivelyTraverseDeletionEffects (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:17482:13)
    at commitDeletionEffectsOnFiber (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:17606:15)
    at recursivelyTraverseDeletionEffects (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:17482:13)
    at commitDeletionEffectsOnFiber (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:17573:15)
    at recursivelyTraverseDeletionEffects (https://localhost:8101/node_modules/.vite/deps/chunk-JZ3YVIXN.js?v=594098e6:17482:13)


The above error occurred in the <Provider> component:

    at e3 (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:220353:19)
    at div
    at FocusRectsProvider (https://localhost:8101/node_modules/.vite/deps/chunk-4GAFNO7A.js?v=594098e6:4014:27)
    at https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:208394:45
    at FluentThemeProvider (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:221729:11)
    at BaseProvider (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:260803:11)
    at div
    at ChatComposite (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:261456:11)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at Chat (https://localhost:8101/src/components/chat/Chat.component.tsx:22:3)
    at ion-card-content
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonCardContent
    at ion-row
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonRow
    at ion-row
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonRow
    at AdvocateChat (https://localhost:8101/src/components/advocate/AdvocateChat.component.tsx:36:20)
    at ion-content
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonContent
    at div
    at PageManager (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23680:5)
    at IonPageInternal (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23729:5)
    at IonPage
    at AdvocateChatPage (https://localhost:8101/src/pages/advocate/AdvocateChat.page.tsx:93:19)
    at Suspense
    at Route2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:772:29)
    at ViewLifeCycleManager (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:24553:5)
    at ion-router-outlet
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonRouterOutlet
    at StackManager (https://localhost:8101/node_modules/.vite/deps/@ionic_react-router.js?v=594098e6:225:5)
    at IonRouterOutletContainer (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23838:5)
    at IonRouterOutlet
    at div
    at ion-tabs
    at div
    at PageManager (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23680:5)
    at <anonymous> (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:24066:5)
    at AdvocateTabs (https://localhost:8101/src/routes/AdvocateTabs.router.tsx:32:21)
    at Suspense
    at Route2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:772:29)
    at PrivateRoute (https://localhost:8101/src/helpers/router.helper.tsx:22:3)
    at Switch2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:916:29)
    at MiddlewareRouter
    at NavManager (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:24720:5)
    at IonRouterInner (https://localhost:8101/node_modules/.vite/deps/@ionic_react-router.js?v=594098e6:484:5)
    at C2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:955:37)
    at Router2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:485:30)
    at IonReactHashRouter (https://localhost:8101/node_modules/.vite/deps/@ionic_react-router.js?v=594098e6:768:5)
    at ion-app
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonApp
    at IonApp (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23644:5)
    at App (https://localhost:8101/src/App.tsx:52:20)
    at PersistGate2 (https://localhost:8101/node_modules/.vite/deps/redux-persist_integration_react.js?v=594098e6:90:5)
    at Provider (https://localhost:8101/node_modules/.vite/deps/react-redux.js?v=594098e6:1173:3)
```

Mounting
```
Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of r3 which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node
    at ul
    at r3 (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:217173:23)
    at https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:219252:16
    at LiveAnnouncer (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:243885:68)
    at r3 (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:217996:28)
    at qo2 (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:218031:17)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at r3 (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:217996:28)
    at qo2 (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:218031:17)
    at MessageThread (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:244085:15)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at ChatScreen (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:261346:11)
    at ChatAdapterProvider (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:260857:11)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at WithBackgroundColor (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:260842:11)
    at Customizer2 (https://localhost:8101/node_modules/.vite/deps/chunk-4GAFNO7A.js?v=594098e6:3295:45)
    at div
    at t5 (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:219820:10)
    at Provider (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:220337:89)
    at e3 (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:220353:19)
    at div
    at FocusRectsProvider (https://localhost:8101/node_modules/.vite/deps/chunk-4GAFNO7A.js?v=594098e6:4014:27)
    at https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:208394:45
    at FluentThemeProvider (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:221729:11)
    at BaseProvider (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:260803:11)
    at div
    at ChatComposite (https://localhost:8101/node_modules/.vite/deps/@azure_communication-react.js?v=594098e6:261456:11)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at div
    at ResultComponent (https://localhost:8101/node_modules/.vite/deps/chunk-6U3BHGZF.js?v=594098e6:4232:68)
    at Chat (https://localhost:8101/src/components/chat/Chat.component.tsx:22:3)
    at ion-card-content
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonCardContent
    at ion-row
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonRow
    at ion-row
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonRow
    at AdvocateChat (https://localhost:8101/src/components/advocate/AdvocateChat.component.tsx:36:20)
    at ion-content
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonContent
    at div
    at PageManager (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23680:5)
    at IonPageInternal (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23729:5)
    at IonPage
    at AdvocateChatPage (https://localhost:8101/src/pages/advocate/AdvocateChat.page.tsx:93:19)
    at Suspense
    at Route2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:772:29)
    at ViewLifeCycleManager (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:24553:5)
    at ion-router-outlet
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonRouterOutlet
    at StackManager (https://localhost:8101/node_modules/.vite/deps/@ionic_react-router.js?v=594098e6:225:5)
    at IonRouterOutletContainer (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23838:5)
    at IonRouterOutlet
    at div
    at ion-tabs
    at div
    at PageManager (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23680:5)
    at <anonymous> (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:24066:5)
    at AdvocateTabs (https://localhost:8101/src/routes/AdvocateTabs.router.tsx:32:21)
    at Suspense
    at Route2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:772:29)
    at PrivateRoute (https://localhost:8101/src/helpers/router.helper.tsx:22:3)
    at Switch2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:916:29)
    at MiddlewareRouter
    at NavManager (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:24720:5)
    at IonRouterInner (https://localhost:8101/node_modules/.vite/deps/@ionic_react-router.js?v=594098e6:484:5)
    at C2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:955:37)
    at Router2 (https://localhost:8101/node_modules/.vite/deps/chunk-FYZAQAMR.js?v=594098e6:485:30)
    at IonReactHashRouter (https://localhost:8101/node_modules/.vite/deps/@ionic_react-router.js?v=594098e6:768:5)
    at ion-app
    at ReactComponent (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23289:7)
    at IonApp
    at IonApp (https://localhost:8101/node_modules/.vite/deps/chunk-JXQRAFHQ.js?v=594098e6:23644:5)
    at App (https://localhost:8101/src/App.tsx:52:20)
    at PersistGate2 (https://localhost:8101/node_modules/.vite/deps/redux-persist_integration_react.js?v=594098e6:90:5)
    at Provider (https://localhost:8101/node_modules/.vite/deps/react-redux.js?v=594098e6:1173:3)
```

**Comment by JamesBurnside** [comment]
@DercilioFontes thanks for raising this, looking into it appears to be caused [here](https://github.com/microsoft/fluentui/blob/3d73007c50ca4008618498a998cdaed5675bbda6/packages/fluentui/react-northstar/src/components/Portal/usePortalBox.ts#L52) when the react n* theme provider is unmounted.

We'll work with the folks at fluent to see what the best resolution is

**Comment by netscout** [comment]
@DercilioFontes i had similar issue when i set toggle rendering on ACS Components(using &&).
I found workaround to set visibility instead of toggle rendering like this:
```typescript
style={{
   visibility: chatAdapter ? "visible" : "hidden",
}}
```

I'm sure your case is same as mine but i hope this helps 😊

**Comment by filip-guider** [comment]
@JamesBurnside Does 1.6.0 now fully support React 18?

**Comment by JamesBurnside** [comment]
We have bumped the peer deps to unblock react 18 usage. However, we haven't fixed the strict mode issue above. If you are using react 18 in your application we recommend disabling strict mode (though strict mode doesn't run in production anyway)

**Comment by yxor** [comment]
any updates on this? (with strict mode)

**Comment by JamesBurnside** [comment]
> any updates on this? (with strict mode)

We are going to remove the dependency (fluent northstar) causing the strict mode issue. This change is non-trivial however, and will take some time to get in. You can track this here #3427.

**Comment by dmceachernmsft** [comment]
Closing as we released version 1.9.0-beta.1 this includes final support for react 18 as we removed the dep for Fluent Northstar. please check it out! 😊

https://www.npmjs.com/package/@azure/communication-react/v/1.9.0-beta.1?activeTab=versions
